include/aspect/particle/property/interface.h

```diff
@@ -536,6 +536,29 @@ namespace aspect
           UpdateFlags
           get_needed_update_flags () const;
 
+          /**
+           * Checks if the particle plugin specified by @p name exists
+           * in this model.
+           */
+          bool
+          pluginname_exists(const std::string &name) const;
```

Comment:
ah, yes. I based this of the `fieldyname_exists` plugin thinking it might be some grammer thing. `plugin_name_exists` seems better to me too. I will change it.